app/Domain/Integrations/Repositories/EloquentIntegrationRepository.php

```diff
@@ -192,4 +196,23 @@ private function saveTransaction(Integration $integration, ?string $couponCode):
             }
         });
     }
+
+    /**
+     * @throws InconsistentIntegrationTypeException
+     */
+    private function guardConsistencyIntegrationType(UuidInterface $subscriptionId, Integration $integration): void
+    {
+        /** @var SubscriptionModel $subscriptionModel */
+        $subscriptionModel = SubscriptionModel::query()->findOrFail($subscriptionId->toString());
```

Comment:
We already inject the `EloquentUdbOrganizerRepository` so I think it's fine to inject the 
`SubscriptionRepository` https://github.com/cultuurnet/publiq-platform/pull/1362/files#diff-d7f8c09ae0d8e31ca6a38cb4d3b676372a2dbaefae83d66772e86285d88a6532L21